framework/tasks/handlers.py

```diff
@@ -48,11 +52,8 @@ def queued_task(task):
     """
     @functools.wraps(task)
     def wrapped(*args, **kwargs):
-        if settings.USE_CELERY:
```

Comment:
Moved to the teardown_request.

This move the USE_CELERY logic to a much more broadly applicable place.